### PR TITLE
Initialization of python variables and type determination

### PR DESCRIPTION
**Pull request by albert-github**
A bit better python initialization (not only the fist "word") and subsequent determination of the type.

### CODE REVIEW

**Comment by doxygen** [comment]
Hi Albert, do you have an example where these changes make a difference?

**Comment by albert-github** [comment]
See in the attachment e.g. the initialization value of `idLen:`  `idLen = '.' * 4`
in the old version it gives:
`string idLen = '.'`
and now:
`string idLen = '.' * 4`


[aa.txt](https://github.com/doxygen/doxygen/files/441069/aa.txt)

File: src/pyscanner.l [hunk 1]
```diff
@@ -1205,51 +1205,49 @@ STARTDOCSYMS      "##"
...
+			if (current->type.isEmpty()) current->type = "string";
 			current->initializer += yytext;
 			g_copyString=&current->initializer;
-       			g_stringContext=VariableEnd;
+       			g_stringContext=VariableDec;
                         BEGIN( DoubleQuoteString );
      		      }
    {TRIDOUBLEQUOTE}   { // start of a comment block
-			current->type = "string";
+			if (current->type.isEmpty()) current->type = "string";
```
Comment (doxygen) [inline src/pyscanner.l]: This makes a """-style comment appear as part of the initializer. I don't think that's right.

File: src/pyscanner.l [hunk 2]
```diff
@@ -1205,51 +1205,49 @@ STARTDOCSYMS      "##"
...
 			g_doubleQuote=TRUE;
 			g_copyString=&current->initializer;
-       			g_stringContext=VariableEnd;
+       			g_stringContext=VariableDec;
 			BEGIN(TripleString);
                       }
 
    {TRISINGLEQUOTE}   { // start of a comment block
-			current->type = "string";
+			if (current->type.isEmpty()) current->type = "string";
```
Comment (doxygen) [inline src/pyscanner.l]: This makes a ''' style comment appear as part of the initializer. I don't think that's right.

**Comment by albert-github** [comment]
I'll have a look at it.

**Comment by albert-github** [comment]
I think it is correct a when we have the attached file the result of the python (2.7.10) command is:
```
D:\>python bb.py
 docu of idStrp  docu of idStrp
.. docu of idStrp
 docu of idStrp  docu of idStrp
.. docu of idStrp
```

and the result of doxygen is:
```
 string 	bb.idStr = """ docu of idStrp """ * 2
 string 	bb.idStrp = '.' + '.' """ docu of idStrp """
 string 	bb.idStr_single = ''' docu of idStrp ''' * 2
 string 	bb.idStrp_single = '.' + '.' ''' docu of idStrp '''
```

[bb.txt](https://github.com/doxygen/doxygen/files/441104/bb.txt)

**Pull request merged**